content/doc/administrate/tcp-redirections.md

```diff
@@ -11,37 +11,93 @@ keywords:
 type: docs
 ---
 {{< callout type="warning" >}}
-TCP redirections are currently available free of charge as long as the feature is still considered in a BETA stage.
+TCP redirections are currently available free of charge as long as the feature is in BETA stage.
 Once the feature leaves the BETA stage, it won't be free anymore.
 {{< /callout >}}
 
 ## What is a TCP redirection?
 
-A regular application on Clever Cloud needs to listen on port 8080 and answer to HTTP traffic.
+Most of Clever Cloud's applications must listen on port 8080 and answer to HTTP traffic.
+This serves requests routing as well as monitoring purposes.
+
+However, some applications might require binary protocol, with raw TCP interactions.
+In that case, you can use a TCP redirection to route TCP traffic from reverse proxies to your application.
+
+The IN port on reverse proxies (public side) is the usual 4040.
+The TCP redirection assigns a random OUT port specifically to your application (starting from port 5000)
+Configure the application to listen to this provided port.
+
+Once done, your application can now send and receive TCP traffic.
+
+
 Our reverse proxies then make your application available using HTTP(s) using your domain name.
 
 Some applications might additionally want to use a binary protocol, with raw TCP interactions.
 For this scenario, your application will have to listen for TCP traffic on port 4040.
 Once this is done, all you need to do is add a TCP redirection to your application. A port will be
-provided to you, and your TCP listener will be available on your domain name using the provided port.
+
 
 ## What is a namespace?
 
-When creating a TCP redirection, you will need to choose in which namespace you want to create it.
-If you are a Premium customer with dedicated reverse proxies, you will have your own namespace.
-If you are not a Premium customer and want to use a custom domain name, you will have to use the
+A namespace is a group of reverse proxies: either the default public ones, cleverapps.io, or dedicated ones
+A TCP redirection is associated with one of those 3 groups.
+
+- If you are  a regular customer and want to use a custom domain name, you have to use the
 "default" namespace.
-If you want to use a cleverapps domain, you will have to use the "cleverapps" namespace.
+- If you are a premium customer with dedicated reverse proxies, you have your own namespace.
+- If you want to use a cleverapps.io domain, you can use the "cleverapps" namespace.
 
-`clever tcp-redirs list-namespaces` will provide you with a list of namespaces that you are allowed
-to use for the current application.
+If you aren't sure what are the possible namespaces for your application, you can use `clever tcp-redirs list-namespaces`.
+This command lists all the possible namespaces current application.
```

Comment:
this sentence lacks a preposition between "possible namespaces" and "current application"